2.train-VAE/scripts/nbconverted/beta_vae_training.py

```diff
@@ -29,56 +28,117 @@
 # In[2]:
 
 
-# load the data 
-data_directory = "../0.data-download/data/"
-dfs = load_data(data_directory, adult_or_pediatric = "all")
-dependency_df = dfs[1]
-sample_df = dfs[0]
+# load the data
+data_directory = pathlib.Path("../0.data-download/data")
+dfs = load_train_test_data(data_directory, train_or_test = "all")
+train_init = dfs[0]
+test_init = dfs[1]
+gene_stats = dfs[2]
 
 
 # In[3]:
 
 
-dependency_df
+# drop the string values
+train_df = train_init.drop(columns= ["DepMap_ID", "age_and_sex"])
+test_df = test_init.drop(columns= ["DepMap_ID", "age_and_sex"])
 
 
 # In[4]:
 
 
-dependency_df1 = dependency_df.drop(axis=0, columns= "DepMap_ID")
-dependency_df1.reset_index(drop=True, inplace=True)
-print(dependency_df1.shape)
-dependency_df1.head(3)
+# subsetting the genes 
+# create dataframe containing the 1000 genes with the largest variances and their corresponding gene label and extract the gene labels
+largest_var_df = gene_stats.nlargest(1000, "variance")
+gene_list = largest_var_df["gene_ID"].tolist()
+gene_list
+
+# create new training and testing dataframes that contain only the corresponding genes
+subset_train_df = train_df.filter(gene_list, axis = 1)
+subset_test_df = test_df.filter(gene_list, axis = 1)
 
 
 # In[5]:
 
 
-encoder_architecture = [250]
-decoder_architecture = [250]
+print(subset_train_df.shape)
+subset_train_df.head(3)
 
 
 # In[6]:
 
 
+print(subset_test_df.shape)
+subset_test_df.head(3)
+
+
+# In[7]:
+
+
+# scale the data
+def absolute_maximum_scale(series):
+    return series / series.abs().max()
+
+
+for col in subset_train_df.columns:
+    subset_train_df[col] = absolute_maximum_scale(subset_train_df[col])
+for col in subset_test_df.columns:
+    subset_test_df[col] = absolute_maximum_scale(subset_test_df[col])
+
+
+# In[8]:
+
+
+encoder_architecture = []
+decoder_architecture = []
+
+
+# In[9]:
+
+
 cp_vae = VAE(
-    input_dim= dependency_df1.shape[1],
-    latent_dim=90,
-    batch_size=32,
-    encoder_batch_norm=True,
-    epochs=58,
-    learning_rate=0.0001,
+    input_dim=subset_train_df.shape[1],
+    latent_dim=10,
+    batch_size=16,
+    encoder_batch_norm=False,
+    epochs=40,
+    learning_rate=0.005,
     encoder_architecture=encoder_architecture,
     decoder_architecture=decoder_architecture,
-    beta=0.06,
+    beta=1.6,
+    lam=0,
     verbose=True,
 )
 
 cp_vae.compile_vae()
 
 
-# In[ ]:
+# In[10]:
+
+
+cp_vae.train(x_train = subset_train_df, x_test = subset_test_df)
+
+
+# In[11]:
+
+
+# display training history
+history_df = pd.DataFrame(cp_vae.vae.history.history)
+history_df
+
+
+# In[12]:
+
 
+# plot and save the figure
+save_path = pathlib.Path('../1.data-exploration/figures/training_curve.png')
```

Comment:
This is great - for when you make the final figure, outputting the training curve data in the cell above is optimal. This way you can modify the visualization without having to re-train each and every time :)